python/graphstorm/dataloading/dataloading.py

```diff
@@ -917,3 +924,175 @@ def __iter__(self):
+
+class DataManager:
+    r"""Data Manager. Combines a file sampler and a dataloader generator,
+    and streamingly provides an iterable over a set of files.
+
+    Parameters:
+    ----------
+        dataloader_generator : DataloaderGenerator: 
+            A dataloader generator
+            Generates dataloader based on given a file path.
+        dataset_path str : 
+            Path to the data files.
+        shuffle : bool
+            Set to ``True`` to have the files reshuffled at every epoch (default: ``False``).
+        infinite : bool
+            Set to ``True`` to have the files sampled infinitely (default: ``False``).
+        local_rank : int
+            Local rank for the trainer (default: ``-1``).
+        is_train : bool
+            Set to ``True`` if the provider is for training set (default: ``True``).
+    """
+
+    def __init__(
+        self,
+        dataloader_generator,
+        dataset_path,
+        shuffle=False,
+        infinite=False,
+        local_rank=-1,
+        world_size=1,
+        is_train=True,
+    ):
+        super().__init__()
+        # TODO (HZ): implement prefetch to put files into a queue
+        self.is_train = is_train
+        if dataset_path is not None:
+            if not isinstance(dataset_path, str):
+                raise TypeError(
+                    "dataset_path should be a str, but got "
+                    "dataset_path={}".format(dataset_path)
+                )
+            file_sampler = DistributedFileSampler(
+                dataset_path=dataset_path,
+                shuffle=shuffle,
+                infinite=infinite,
+                local_rank=local_rank,
+                world_size=world_size,
+                is_train=is_train,
+            )
+        else:
+            raise ValueError(
+                    "dataset_path needs to be specified"
+                )
+        self.file_sampler = file_sampler
+        self.file_sampler_iter = iter(self.file_sampler)
+
+        self.dataloader_generator = dataloader_generator
+
+        self.dataloader = None
+        self.data_file = None
+
+        logging.info(f"DataProvider - Initialization: num_files = {len(file_sampler)}")
+
+    def get_iterator_name(self):
+        """ Return shard name.
+        """
+        return self.data_file
+
+    def get_iterator(self):
+        """ Get dataloader iterator from a data file.
+        """
+        dataloader = None
+
+        data_file = next(self.file_sampler_iter)
+
+        # TODO (HZ): implement a queue to store and pop the files by prefetch and get_iterator
+
+        if data_file is not None:
+            dataloader = self.dataloader_generator.generate(data_file, is_train=self.is_train)
+
+        self.data_file = data_file
+        self.dataloader = dataloader
+
+        return dataloader
+
+    def release_iterator(self):
+        """ Release the dataloader iterator.
+        """
+        del self.dataloader
```

Comment:
You just need to set self.dataloader to None here.

Reply:
addressed by new commit.